exercises/forth/test/Tests.hs

```diff
@@ -1,74 +1,78 @@
 {-# LANGUAGE OverloadedStrings #-}
-import Test.HUnit (Assertion, (@=?), runTestTT, Test(..), Counts(..))
-import System.Exit (ExitCode(..), exitWith)
-import Forth (ForthError(..), evalText, empty, formatStack)
-import Control.Monad (foldM)
-import Data.Text (Text)
 
-exitProperly :: IO Counts -> IO ()
-exitProperly m = do
-  counts <- m
-  exitWith $ if failures counts /= 0 || errors counts /= 0 then ExitFailure 1 else ExitSuccess
+import Control.Monad     (foldM)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
 
-testCase :: String -> Assertion -> Test
-testCase label assertion = TestLabel label (TestCase assertion)
+import Forth (ForthError(..), empty, evalText, formatStack)
 
 main :: IO ()
-main = exitProperly $ runTestTT $ TestList
-       [ TestList forthTests ]
-
-runTexts :: [Text] -> Either ForthError Text
-runTexts xs = formatStack <$> foldM (flip evalText) empty xs
-
-forthTests :: [Test]
-forthTests =
-  [ testCase "no input, no stack" $
-    "" @=? formatStack empty
-  , testCase "numbers just get pushed onto the stack" $
-    Right "1 2 3 4 5" @=? runTexts ["1 2 3 4 5"]
-  , testCase "non-word characters are separators" $
-    -- Note the Ogham Space Mark ( ), this is a spacing character.
```

Comment:
> I think that, ideally, people should get all the information needed to solve an exercise in the following order:

This seems to make sense; I think it represents how we would interact with a library we are considering to use - we either run the tests or read its documentation (some people switch the order of these two don't they) and only as the resort will they read the code.

When I am solving an exercise I tend to look quite often at the tests (so I know what's coming up next and I know what's expected of me), but I can't speak for everyone.

I think my main question is - now that this line appears in the README, when someone looks at that, what do you think their reaction will be? Will it be "I'll immediately search for all instances of this   character to see what this is all about", or will it be "Why are you telling me this? I will forget about it". Well, it's not my place to predict how people will act.

> An alternative would be to improve the tests and descriptions, if you think they are not good enough.

I'm not sure if "non-word characters are separators, including the Ogham Space Mark ( )" would be too much detail...

I question the case because visually the Ogham space mark is very easy to mistake for a minus sign, so someone looking at this might think this is five minus six, rather than five space six.

Perhaps I will take solace in the fact that most people probably don't fail this test because `isSpace` will do the right thing.